command/genesis.go

```diff
@@ -228,26 +241,55 @@ func (c *GenesisCommand) Run(args []string) int {
 		Bootnodes: bootnodes,
 	}
 
-	if len(premine) != 0 {
-		for _, prem := range premine {
-
-			var addr types.Address
-			val := defaultPremineBalance
-			if indx := strings.Index(prem, ":"); indx != -1 {
-				// <addr>:<balance>
-				addr, val = types.StringToAddress(prem[:indx]), prem[indx+1:]
-			} else {
-				// <addr>
-				addr = types.StringToAddress(prem)
-			}
+	for _, prem := range premine {
+		var addr types.Address
+		val := defaultPremineBalance
+		if indx := strings.Index(prem, ":"); indx != -1 {
+			// <addr>:<balance>
+			addr, val = types.StringToAddress(prem[:indx]), prem[indx+1:]
+		} else {
+			// <addr>
+			addr = types.StringToAddress(prem)
+		}
 
-			amount, err := types.ParseUint256orHex(&val)
-			if err != nil {
-				c.UI.Error(fmt.Sprintf("failed to parse amount %s: %v", val, err))
-				return 1
-			}
+		amount, err := types.ParseUint256orHex(&val)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("failed to parse amount %s: %v", val, err))
+			return 1
+		}
+		cc.Genesis.Alloc[addr] = &chain.GenesisAccount{
+			Balance: amount,
+		}
+	}
+
+	for _, pres := range prestake {
+		var addr types.Address
+		val := defaultPrestakeBalance
+		if indx := strings.Index(pres, ":"); indx != -1 {
+			// <addr>:<balance>
+			addr, val = types.StringToAddress(pres[:indx]), pres[indx+1:]
+		} else {
+			// <addr>
+			addr = types.StringToAddress(pres)
+		}
+
+		stakeAmount, err := types.ParseUint256orHex(&val)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("failed to parse stake amount %s: %v", val, err))
+			return 1
+		}
+
+		previousAccount := cc.Genesis.Alloc[addr]
+
+		if previousAccount != nil {
+			// Account already has a premined balance
+			previousAccount.StakedBalance = stakeAmount
+			cc.Genesis.Alloc[addr] = previousAccount
```

Comment:
Maybe this line is not needed since `previousAccount` is a pointer to the `GenesisAccount`?
```suggestion
			previousAccount.StakedBalance = stakeAmount
```

Reply:
Agreed. Removing now 👍